### ISSUE TITLE
proc macro panicked CompositeTemplate

### ISSUE THREAD

**Issue by Tibladar** [ilya-zlobintsev/LACT#211]
When compiling from master I get the following errors:

```
   Compiling gtk4 v0.7.3
   Compiling lact-gui v0.5.0 (/home/user/LACT/lact-gui)
error: proc macro panicked
  --> lact-gui/src/app/info_row.rs:32:14
   |
32 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 5 column 10:
              5 |template $InfoRow: Box {
                |         ^
           
error: proc macro panicked
  --> lact-gui/src/app/page_section.rs:37:14
   |
37 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $PageSection: Box {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

```

<details><summary>Full logs with more errors like above</summary>

```
   Compiling gtk4 v0.7.3
   Compiling lact-gui v0.5.0 (/home/user/LACT/lact-gui)
error: proc macro panicked
  --> lact-gui/src/app/info_row.rs:32:14
   |
32 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 5 column 10:
              5 |template $InfoRow: Box {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/page_section.rs:37:14
   |
37 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $PageSection: Box {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/root_stack/info_page/vulkan_info/feature_window/row.rs:31:14
   |
31 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $VulkanFeatureRow: Box {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/root_stack/info_page/vulkan_info/feature_window/mod.rs:38:14
   |
38 |     #[derive(CompositeTemplate, Properties, Default)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $VulkanFeaturesWindow: Window {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/root_stack/oc_page/gpu_stats_section.rs:85:14
   |
85 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $GpuStatsSection: $PageSection {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/root_stack/oc_page/power_cap_section.rs:44:14
   |
44 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $PowerCapSection: $PageSection {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/root_stack/oc_page/power_states/power_state_row.rs:37:14
   |
37 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $PowerStateRow: Box {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/root_stack/oc_page/power_states/power_states_frame.rs:80:14
   |
80 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $PowerStatesFrame: Box {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/root_stack/oc_page/power_states/power_states_list.rs:94:14
   |
94 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $PowerStatesList: Frame {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: proc macro panicked
  --> lact-gui/src/app/root_stack/software_page.rs:48:14
   |
48 |     #[derive(CompositeTemplate, Default, Properties)]
   |              ^^^^^^^^^^^^^^^^^
   |
   = help: message: called `Result::unwrap()` on an `Err` value: error: Could not determine what kind of syntax is meant here
           at <stdin> line 4 column 10:
              4 |template $SoftwarePage: Box {
                |         ^
           
           
   = note: this error originates in the derive macro `CompositeTemplate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `lact-gui` (lib) due to 10 previous errors
```

</details>

The errors look like #210 but I here's no mention of blueprint-compiler and I have that installed.

Ubuntu 22.04
rustc 1.70.0 (90c541806 2023-05-31) 
cargo 1.70.0

**Comment by ilya-zlobintsev** [comment]
The blueprint-compiler version shipped with ubuntu 22.04 is too old. You can install a deb file of a newer version, here's how it's done for the package builds: https://github.com/ilya-zlobintsev/LACT/blob/master/pkg/recipes/lact/recipe.yml#L30

**Comment by ilya-zlobintsev** [comment]
Done: https://github.com/ilya-zlobintsev/LACT/commit/7e2791e79ae4fbbd587780d0c7766acb89adbe9d
